rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1251,20 +1251,30 @@ private void processReleaseTransactions(Transaction rskTx) {
                 bridgeConstants.getRsk2BtcMinimumAcceptableConfirmations(),
                 Optional.of(1)
         );
-        if (!txsWithEnoughConfirmations.isEmpty()) {
-            ReleaseTransactionSet.Entry entry = txsWithEnoughConfirmations.iterator().next();
-            // Since RSKIP176 we are moving back to using the updateCollections related txHash as the set key
-            if (activations.isActive(ConsensusRule.RSKIP146) && !activations.isActive(ConsensusRule.RSKIP176)) {
-                // The release transaction may have been created prior to the Consensus Rule activation
-                // therefore it won't have a rskTxHash value, fallback to this transaction's hash
-                txsWaitingForSignatures.put(entry.getRskTxHash() == null ? rskTx.getHash() : entry.getRskTxHash(), entry.getTransaction());
-            }
-            else {
-                txsWaitingForSignatures.put(rskTx.getHash(), entry.getTransaction());
-            }
-            if(activations.isActive(ConsensusRule.RSKIP326)) {
-                eventLogger.logPegoutConfirmed(entry.getTransaction().getHash(), entry.getRskBlockNumber());
-            }
+        if (txsWithEnoughConfirmations.isEmpty()) {
+            return;
+        }
+
+        ReleaseTransactionSet.Entry entry = txsWithEnoughConfirmations.iterator().next();
+
+        Keccak256 rskTxHash;
+        if (activations.isActive(ConsensusRule.RSKIP375)){
+            rskTxHash = entry.getRskTxHash();
+        }
+        // Since RSKIP176 we are moving back to using the updateCollections related txHash as the set key
+        else if (activations.isActive(ConsensusRule.RSKIP146) && !activations.isActive(ConsensusRule.RSKIP176)) {
+            // The release transaction may have been created prior to the Consensus Rule activation
+            // therefore it won't have a rskTxHash value, fallback to this transaction's hash
+            rskTxHash = entry.getRskTxHash() == null ? rskTx.getHash() : entry.getRskTxHash();
+        }
+        else {
+            rskTxHash = rskTx.getHash();
+        }
```

Comment:
What do you think about this logic to it's own private method? Something like `getTxsWaitingForSignatureKey`